motion_control/drive_controller/include/drive_controller/DifferentialDriveControllerParameters.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2024 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/// @brief       Differential drive controller
+/// @author      Mathis Lécrivain <lecrivain.mathis@gmail.com>
+
+#pragma once
+
+namespace cogip {
+
+namespace drive_controller {
+
+class DifferentialDriveControllerParameters {
+public:
+    /// @brief Construct a new DifferentialDriveControllerParameters object
+    /// @param left_wheel_diameter left wheel diameter value (mm)
+    /// @param right_wheel_diameter right wheel diameter value (mm)
+    /// @param track_width Track width value (mm)
+    /// @param left_motor_constant left motor constant (no unit)
+    /// @param right_motor_constant right motor constant (no unit)
+    /// @param max_speed_percentage max speed ration to send to motors (%) in range [0;100]
+    /// @param loop_period regulation loop period (ms)
+    DifferentialDriveControllerParameters(double left_wheel_diameter,
+                                          double right_wheel_diameter,
+                                          double track_width,
+                                          double left_motor_constant,
+                                          double right_motor_constant,
+                                          double max_speed_percentage,
+                                          double loop_period)
+                                          : left_wheel_diameter_(left_wheel_diameter),
+                                            right_wheel_diameter_(right_wheel_diameter),
+                                            track_width_(track_width),
+                                            left_motor_constant_(left_motor_constant),
+                                            right_motor_constant_(right_motor_constant),
+                                            max_speed_percentage_(max_speed_percentage),
+                                            loop_period_(loop_period) {};
+
+    /// @brief Set the left motor wheel diameter dimension
+    /// @param wheel_diameter Wheel diameter value (mm)
+    void set_left_wheel_diameter(double wheel_diameter) { left_wheel_diameter_ = wheel_diameter; }
```

Comment:
Do you think a wheel diameter could change ? :thinking:

Reply:
It will be more probable for the odometer wheel size but in case of the encoder mounted on the motor wheel, we can change it according to odometry calibration results too.

Reply:
true and it could be useful to keep setters to keep firmware generic and being setup only by the RPi.